packaging/tags.py

```diff
@@ -225,10 +225,33 @@ def cpython_tags(
                 yield Tag(interpreter, "abi3", platform_)
 
 
-def _generic_abi() -> Iterator[str]:
-    abi = sysconfig.get_config_var("SOABI")
-    if abi:
-        yield _normalize_string(abi)
+def _generic_abi() -> List[str]:
+    """Return the ABI tag based on EXT_SUFFIX"""
+    # ext_suffix is something like this. We want to keep the parts which are
+    # related to the ABI and remove the parts which are related to the
+    # platform:
+    # - linux:   '.cpython-310-x86_64-linux-gnu.so' => cp310
+    # - mac:     '.cpython-310-darwin.so'           => cp310
+    # - win:     '.cp310-win_amd64.pyd'             => cp310
+    # - pypy:    '.pypy38-pp73-x86_64-linux-gnu.so' => pypy38_pp73
+    # - graalpy: '.graalpy-38-native-x86_64-darwin.dylib'
+    #                                               => graalpy_38_native
+
+    ext_suffix = _get_config_var("EXT_SUFFIX", warn=True)
+    if not isinstance(ext_suffix, str) or ext_suffix[0] != ".":
+        raise SystemError("invalid sysconfig.get_config_var('EXT_SUFFIX')")
```

Comment:
Is this correct or would the reviewers prefer the function return `[]` for an invalid `EXT_SUFFIX` (which should never happen)? I prefer exceptions when reality clashes with my expectations, but maybe that is too optimistic?

Reply:
If you can't function w/o the information then an exception makes sense (have not thought about which one).